command/server.go

```diff
@@ -1809,6 +1821,49 @@ func (c *ServerCommand) Run(args []string) int {
 				pFile.Close()
 			}
 
+			c.logger.Info(fmt.Sprintf("Wrote pprof files to: %s", pprofPath))
+		case <-heapProfCh:
+			path := c.flagHeapProfile
+			if _, err := os.Stat(path); err != nil {
```

Comment:
Isn't this going to error if the path to save the files in doesn't exist, which means we never make it to line 1832 below, where you're creating the new directory?

Off the cuff, it feels like we need 3 different branches of logic here:
- err is nil, meaning the path exists, so that's where we put the files
- err is not nil but the error is ErrNotExist, which is fine, and not actually an error case for us, in which case we need to create the directory
- err is not nil and is also not ErrNotExist, meaning there's some legitimate error that will prevent this process from continuing, in which case we need to exit early.

Am I misunderstanding this?

Reply:
Added a check for `ErrNotExist`, I was previously lumping case 2 and 3 together.